source/containers/DynamicCardRenderer/DynamicCardRenderer.tsx

```diff
@@ -156,62 +171,46 @@ const renderCardComponent = (
           circle={component.circle}
         />
       );
-  }
-
-  // Treat buttons separately, because they have some more complicated behavior
-  if (component.type === "button") {
-    const { icon, iconPosition, text } = component;
-    const onClick: () => void = () => null;
+    case "button":
+      if (component.action === "infoModal") {
+        return (
+          <InfoModalButton key={`${index}-${component.type}`} {...component} />
+        );
+      }
 
-    // treat info-modal separately since it doesn't fit the same pattern as the other buttons.
-    if (component.action === "infoModal") {
       return (
-        <InfoModalButton key={`${index}-${component.type}`} {...component} />
+        <Card.Button
+          key={`${index}-${component.type}`}
+          onClick={handleClick(component, navigation)}
+        >
+          {component.icon &&
+            component.iconPosition &&
+            component.iconPosition === "left" && <Icon name={component.icon} />}
+          <TextComponent>{component.text}</TextComponent>
+          {component.icon &&
+            (!component.iconPosition || component.iconPosition === "right") && (
+              <Icon name={component.icon} />
+            )}
+        </Card.Button>
       );
-    }
-
-    return (
-      <Card.Button
-        key={`${index}-${component.type}`}
-        onClick={handleClick(component, navigation)}
-      >
-        {icon && iconPosition && iconPosition === "left" && (
-          <Icon name={icon} />
-        )}
-        <TextComponent>{text}</TextComponent>
-        {icon && (!iconPosition || iconPosition === "right") && (
-          <Icon name={icon} />
-        )}
-      </Card.Button>
-    );
+    default:
+      return null;
   }
 };
 
-interface Props {
-  colorSchema?: "blue" | "red" | "green" | "purple" | "neutral";
-  backgroundColor?: "blue" | "red" | "green" | "purple" | "neutral";
-  shadow?: boolean;
-  outlined?: boolean;
-  components: CardComponent[];
-}
-
-/** Dynamically renders a card with the sent in children as an array of json objects. */
 const DynamicCardRenderer: React.FC<Props> = ({
   colorSchema,
   backgroundColor,
   shadow,
   outlined,
   components,
+  completionsClarification,
 }) => {
-  let navigation: any = {};
+  let navigation: unknown = {};
 
   const { user } = useContext(AuthContext);
+  navigation = useNavigation();
```

Comment:
Not your code but...

```js
const navigation = useNavigation();
```

instead of:

```js
let navigation: unknown = {};
navigation = useNavigation();
```